### PR TITLE
CLDR-15296 Add NIST units

### PR DESCRIPTION
**Pull request by macchiati**
CLDR-15296

This is preparation for adding more units (eg Japanese units of length), and adding NIST quantities.

common/supplemental/units.xml
- added -it for BTU, calorie variants
- changed comment to description
- fixed 'dose' to be more general
- added some english units and NIST units
- ignore the commented-out abhenry unit alias

common/validity/unit.xml
- added new long units

tools/cldr-code/src/main/java/org/unicode/cldr/util/DtdData.java
- Changed the MapComparator<String> unitOrder to be data-driven. 
- Ignore the commented out old stuff for this PR
- Changed raw access to unitOrder to be a function getUnitOrder(), allowing for lazy loading (needed to avoid circularity!)

tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java
- Changed raw access to unitOrder to be a function getUnitOrder()

tools/cldr-code/src/main/java/org/unicode/cldr/util/Rational.java
- Minor changes to allow for longer repeating format

tools/cldr-code/src/main/java/org/unicode/cldr/util/UnitConverter.java
- Better exception message
- Added pow4 (needed for NIST)
- Added kilogram-force to the SKIP_PREFIX set of exceptions

tools/cldr-code/src/main/java/org/unicode/cldr/util/UnitParser.java
- added method to get left-over items (used in test)
- fixed problem with type of value

tools/cldr-code/src/test/java/org/unicode/cldr/unittest/ExternalUnitConversionData.java
tools/cldr-code/src/test/java/org/unicode/cldr/unittest/NistUnits.java
- Added information about the source ('from')
- Added handling of BTU IT and calorie IT

tools/cldr-code/src/test/java/org/unicode/cldr/unittest/GenerateNewUnits.java
- Added tool to make it easier to add new units based on data files. For now, focused on NIST
- This replaces a hack in TestUnits at around 211

tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestUnits.java
- Broke out the debugging flags to make more granular
- Added simple TestAUnits() for displaying information (if isVerbose)
- Around line 330, just formatting differences. Easiest to Hide Whitespace in the diff.
- Added quantityToCategory and CATEGORY_EXCEPTIONS, so that we could do a more granular test, and get rid of a very ugly set of switches, at around 767 below.
- Added BTU IT and calorie IT to STILL_RECOGNIZED_SIMPLES
- Added new test TestValidityAgainstUnitFile
- Around L2714, added test to skip over printing additional failure cases, to make the test faster.
- Around L3123, dropped a condition that shouldn't have been tested.

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Comment by srl295** [comment]
@macchiati  sub-PR https://github.com/macchiati/cldr/pull/62 to fix formatting

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestUnits.java [hunk 1]
```diff
@@ -1,28 +1,5 @@
 package org.unicode.cldr.unittest;
 
-import com.google.common.base.Joiner;
```
Comment (macchiati) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestUnits.java:3]: @srl295 I think the Eclipse formatter is fighting a battle with spotless. So have to figure out how to disable that.

Comment (macchiati) [reply tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestUnits.java:3]: Also, the maven build is having trouble: Error:  Failed to execute goal org.apache.maven.plugins:maven-resources-plugin:3.3.1:resources (default-resources) on project cldr-code: Execution default-resources of goal org.apache.maven.plugins:maven-resources-plugin:3.3.1:resources failed: Plugin org.apache.maven.plugins:maven-resources-plugin:3.3.1 or one of its dependencies could not be resolved: Failed to collect dependencies at org.apache.maven.plugins:maven-resources-plugin:jar:3.3.1 -> org.codehaus.plexus:plexus-interpolation:jar:1.26: Failed to read artifact descriptor for org.codehaus.plexus:plexus-interpolation:jar:1.26: Could not transfer artifact org.codehaus.plexus:plexus-interpolation:pom:1.26 from/to central (https://repo.maven.apache.org/maven2): transfer failed for https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-interpolation/1.26/plexus-interpolation-1.26.pom: Unknown host repo.maven.apache.org: Name or service not known -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <args> -rf :cldr-code
Error: Process completed with exit code 1.

**Comment by srl295** [comment]
Yes you can manually run spotless using the procedure in todays CLDR
agenda.

As to plexus sounds like a temporary network issue. Should resolve and then
be cached

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/ExternalUnitConversionData.java [hunk 2]
```diff
@@ -23,18 +23,24 @@ final class ExternalUnitConversionData {
...
+            String from,
             String line,
             Multimap<String, String> changes) {
         super();
         this.quantity = quantity;
 
+        if (source.contains("thermal")) {
+            int debug = 0;
+        }
+
```
Comment (srl295) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/ExternalUnitConversionData.java:43]:
```suggestion
```

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/GenerateNewUnits.java [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+                            + data.source
+                            + "'"
+                            + " baseUnit='"
+                            + data.target
+                            + "'"
+                            + " factor='"
+                            + data.info.factor
+                            + "'"
+                            + " systems=\'metric si\'"
+                            + "/>");
```
Comment (srl295) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/GenerateNewUnits.java:128]: String.format will autoformat better

Comment (macchiati) [reply tools/cldr-code/src/test/java/org/unicode/cldr/unittest/GenerateNewUnits.java:128]: Will follow up with these in later PR

**Pull request merged**